plugins/transcend/transcend-nvme.c

```diff
@@ -5,82 +5,352 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <inttypes.h>
+#include <string.h>
+#include <time.h>
 
 #include "nvme.h"
 #include "libnvme.h"
 #include "plugin.h"
+#include "transcend-nvme.h"
+#include "common.h"
 
 #define CREATE_CMD
-#include "transcend-nvme.h"
+
+#define PLP_RECORD_PATH "PLPRec.txt"
+#define PLP_DATA_EXPIRED_TIME 20
 
 static const __u32 OP_BAD_BLOCK = 0xc2;
 static const __u32 DW10_BAD_BLOCK = 0x400;
 static const __u32 DW12_BAD_BLOCK = 0x5a;
 
-static int getHealthValue(int argc, char **argv, struct command *cmd, struct plugin *plugin)
+static const __u32 OP_HEALTH = 0xc2;
+static const __u32 DW10_HEALTH = 0x01;
+static const __u32 DW12_HEALTH = 0xb3;
+
+static const __u32 OP_ID = 0x06;
+static const __u32 DW10_ID = 0x01;
+
+static const int i_dis = 20;
+static const double full_value = 170;
+
+enum plp_error_code {
+	PLP_ERROR_NO_MATCH = -1,
+	PLP_ERROR_DATA_EXPIRED  = -2
+};
+
+static const char *string_list[] = {
+    "UTE210T",   "MTE712P",   "MTE560P",   "MTE662P", "MTE662P-I", "MTS970P",
+    "MTS952P",   "MTS952P-I", "MTS570P",   "MTS400P", "SSD910T",   "SSD470P",
+    "SSD470P-I", "SSD452P",   "SSD452P-I", "SSD420P", "MSA470P",   "MSA452P"
+};
+
+static const int list_size = ARRAY_SIZE(string_list);
+
+static int get_plp_health(int argc, char **argv, struct command *cmd, struct plugin *plugin);
+static int read_useful_plp_value(const char *device);
+static void record_plp_value(const char *device, int value, bool is_replace);
+
+static int get_health_value(int argc, char **argv, struct command *cmd, struct plugin *plugin)
+{
+    struct nvme_smart_log smart_log;
```

Comment:
These are also indentation errors.

Reply:
FYI: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/coding-style.rst?h=v6.12-rc6#n18

Reply:
Thanks for your quick reply, I will correct it.

Reply:
checkpatch.pl is helpful to detect these kind of formatting issues. You can run it locally. Fetch the linux kernel source and run `linux/scripts/checkpatch.pl -g HEAD` or similar. Keep in mind checkpatch.pl is very opinionated and we should just be ignored some of the advice.